pkg/tests/rabbitmq/rabbitmq_fanout_test.go

```diff
@@ -0,0 +1,178 @@
+package rabbitmq
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/ormushq/ormus/pkg/broker/messagebroker"
+	"github.com/ormushq/ormus/pkg/broker/rabbitmq"
+)
+
+// Define a struct to hold parameters for the fanout test case
+type FanOutTestCase struct {
+	Name           string
+	Mode           string
+	ExchangeName   []string
+	QueueNames     []string
+	NumMessages    int
+	ExpectedCounts map[string]int
+	Expected       int
+}
+
+func TestFanoutMessaging(t *testing.T) {
+	// Define test cases
+	testCases := []FanOutTestCase{
+		{
+			Name:         "same exchange",
+			Mode:         "fanout",
+			ExchangeName: []string{"fanout_exchange", "fanout_exchange", "fanout_exchange", "fanout_exchange"},
+			QueueNames:   []string{"queue1", "queue2", "queue3", "queue4"},
+			NumMessages:  100,
+			ExpectedCounts: map[string]int{
+				"queue1": 100,
+				"queue2": 100,
+				"queue3": 100,
+				"queue4": 100,
+			},
+			//! expected = number of message * number of queue = 400 => for all queue :if auto-delete be false
+			//! and the queue exist before =1600 and 1000 if auto-delete be true
+			Expected: 1000,
+		},
+		{
+			Name:         "different exchange",
+			Mode:         "fanout",
+			ExchangeName: []string{"fanout_exchange1", "fanout_exchange2", "fanout_exchange3", "fanout_exchange4"},
+			QueueNames:   []string{"queue1", "queue2", "queue3", "queue4"},
+			NumMessages:  100,
+			ExpectedCounts: map[string]int{
+				"queue1": 100,
+				"queue2": 100,
+				"queue3": 100,
+				"queue4": 100,
+			},
+			Expected: 400,
+		},
+	}
+	fmt.Println("start tests...")
```

Comment:
Please remove Prints after debugging this makes CI test page messy and out of hand.